Modules/Tokens/test/Tokens.API.Tests.Integration/StepDefinitions/TokensApiStepDefinitions.cs

```diff
@@ -0,0 +1,301 @@
+
+        var httpResponse = await _tokensApi.CreateToken(requestConfiguration);
+        _tokenResponse = httpResponse;
+    }
+
+    [When(@"a GET request is sent to the Tokens endpoint with a list containing t\.Id, p\.Id")]
+    public async Task WhenAGETRequestIsSentToTheTokensEndpointWithAListContainingT_IdP_Id()
+    {
+        var tokenIds = new List<string> { _tokenId, _peerTokenId };
+        var response = await _tokensApi.GetTokenById(_requestConfiguration, tokenIds);
+
+        _tokenResponse.StatusCode = response.StatusCode;
+
+        var tokens = response.Content!.Result!;
+
+        _responseTokens.AddRange(tokens);
+    }
+
+    [Then(@"the response contains both Tokens")]
+    public void ThenTheResponseOnlyContainsTheOwnToken()
+    {
+        _responseTokens.Should().HaveCount(2)
+            .And.Contain(token => token.Id == _tokenId)
+            .And.Contain(token => token.Id == _peerTokenId);
+    }
+
+
+    [Then(@"the response contains all Tokens with the given ids")]
+    public void ThenTheResponseContainsAllTokensWithTheGivenIds()
+    {
+        _responseTokens.Select(t => t.Id)
+            .Should()
+            .HaveCount(_givenOwnTokens.Count)
+            .And.BeEquivalentTo(_givenOwnTokens.Select(t => t.Id), options => options.WithoutStrictOrdering());
+    }
+
+    [Then(@"the response content includes an error with the error code ""([^""]+)""")]
+    public void ThenTheResponseContentIncludesAnErrorWithTheErrorCode(string errorCode)
+    {
+        _tokenResponse.Content!.Error.Should().NotBeNull();
+
+        _tokenResponse.Content!.Error!.Code.Should().Be(errorCode);
+    }
+
+    [Then(@"the response contains a Token")]
+    public void ThenTheResponseContainsAToken()
+    {
+        _tokenResponse.Should().NotBeNull();
+
+        AssertStatusCodeIsSuccess();
+
+        AssertResponseContentTypeIsJson();
+
+        if (_tokenResponse.HttpMethod == Method.Get.ToString())
+        {
+            AssertResponseContentCompliesWithSchema<Token>();
+        }
+        else if (_tokenResponse.HttpMethod == Method.Post.ToString())
+        {
+            AssertResponseContentCompliesWithSchema<CreateTokenResponse>();
+        }
+    }
+
+    [Then(@"the response status code is (\d+) \(.+\)")]
+    public void ThenTheResponseStatusCodeIs(int expectedStatusCode)
+    {
+        var actualStatusCode = (int)_tokenResponse.StatusCode;
+        actualStatusCode.Should().Be(expectedStatusCode);
+    }
+
+    private void AssertStatusCodeIsSuccess()
+    {
+        _tokenResponse.IsSuccessStatusCode.Should().BeTrue();
+    }
+
+    private void AssertResponseContentTypeIsJson()
+    {
+        _tokenResponse.ContentType.Should().Be("application/json");
+    }
+
+    private void AssertResponseContentCompliesWithSchema<T>()
+    {
+        JsonValidators.ValidateJsonSchema<ResponseContent<T>>(JsonConvert.SerializeObject(_tokenResponse.Content!), out var errors)
```

Comment:
Hmm, I think I see why Diogo had the raw content property. By deserializing and re-serializing, the JSON is distorted. 

So I suggest to add the property again - with a different name though. I'd suggest `RawContent`